src/board.cpp

```diff
@@ -0,0 +1,190 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include "board.h"
+#include "cpu.h"
+
+
+Board::Board(GUI *gui, int board_size) : gui_(gui), board_size_(board_size) {
+  num_humans_ = 0;
+}
+
+void Board::init(unsigned int board_size) {
+  if (board_size > 0) {
+    board_size_ = board_size;
+  }
+  for (int i = 0; i < num_players(); ++i) {
+    // Player ids are 1-indexed
+    players_[i]->init(i + 1, board_size_);
+  }
+}
+
+void Board::reset() {
+  for (Player* player : players_) {
+    delete player;
+  }
```

Comment:
ayo you're doing your own memory management that's insane. Okay fine. My comment earlier you'll need `unique_ptr`s

Reply:
oof lol. So unique ptr will auto-delete the pointers when they are removed from the vector then right?